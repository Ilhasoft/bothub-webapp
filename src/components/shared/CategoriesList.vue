<template>
  <div class="categories-list">
    <b-field class="categories-list__field">
      <b-autocomplete
        v-model="category"
        :keep-first="keepFirst"
        :open-on-focus="openOnFocus"
        :data="optionsCategories"
        :placeholder="$t('webapp.home.all_categories')"
        dropdown-position="bottom"
        size="normal"
        @select="option => select(option)" />
    </b-field>
  </div>
</template>

<script>
import { mapActions } from 'vuex';


export default {
  name: 'CategoriesList',
  data() {
    return {
      val: 0,
      allCategories: [],
      keepFirst: true,
      openOnFocus: true,
      category: '',
      selected: null,
    };
  },
  computed: {
    filterCategories() {
      return this.allCategories.filter(option => option.name
        .toString()
        .toLowerCase()
        .indexOf(this.category.toLowerCase()) >= 0);
    },
    optionsCategories() {
      return this.filterCategories.map(option => option.name);
    },
    queryCategoryId() {
      return this.$route.query.category
        ? parseInt(this.$route.query.category || 0, 10)
        : 0;
    },
  },
  watch: {
    val(value) {
      this.$emit('input', value);
    },
    $route() {
      this.val = this.queryCategoryId;
    },
    category() {
      if (this.category === '') {
        this.val = 0;
      }
    },
  },
  async mounted() {
    const categoriesResponse = await this.getAllCategories();
    this.allCategories = categoriesResponse.data;
    this.val = this.queryCategoryId;
  },
  methods: {
    ...mapActions([
      'getAllCategories',
    ]),
    select(option) {
      this.allCategories.filter((item) => {
        if (item.name === option) {
          this.selected = option;
          this.val = item.id;
        }
        return '';
      });
    },
  },
};
</script>

<style lang="scss">
.categories-list{
  margin-top: 1rem;
  display: flex;
  justify-content: flex-end;
  align-items: center;

  &__field{
    width: 12.5rem;
  }
}

</style>
