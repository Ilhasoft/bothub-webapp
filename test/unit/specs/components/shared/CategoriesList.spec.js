import BH from 'bh';
import { shallowMount, createLocalVue } from '@vue/test-utils';
import CategoriesList from '@/components/shared/CategoriesList';


const localVue = createLocalVue();
localVue.use(BH);

describe('CategoriesList.vue', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallowMount(CategoriesList, {
      localVue,
      mocks: {
        $t: () => 'some specific text',
      },
    });
  });

  test('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
