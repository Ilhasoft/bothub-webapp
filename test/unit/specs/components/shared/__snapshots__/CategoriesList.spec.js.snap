// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoriesList.vue renders correctly 1`] = `
<div class="categories-list">
  <b-field-stub addons="true" class="categories-list__field">
    <b-autocomplete-stub value="" data="" field="value" keepfirst="true" openonfocus="true" dropdownposition="bottom" placeholder="some specific text" size="normal"></b-autocomplete-stub>
  </b-field-stub>
</div>
`;
